dll/win32/shell32/shlexec.cpp

```diff
@@ -802,6 +802,15 @@ static UINT SHELL_FindExecutable(LPCWSTR lpPath, LPCWSTR lpFile, LPCWSTR lpVerb,
         xlpFile[0] = '\0';
     }
 
+    /* Clear any trailing periods */
```

Comment:
Like in the remove-blanks case, this could be moved only in the if-body where the xlpFile is known to not be empty.

Reply:
I agree with moving the code up into the if-body as you suggest. This way the ```lpResult``` can benefit from these changes.